cmd/root_commands.go

```diff
@@ -131,3 +158,77 @@ func initCustomiseMyData(cfg *config.Config) *cobra.Command {
 		},
 	}
 }
+
+func generateRepository() *cobra.Command {
+	return &cobra.Command{
+		Use:   "github",
+		Short: "Creates a github hosted repository",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+			nameOfApp, _ := cmd.Flags().GetString("name")
+			token, _ := cmd.Flags().GetString("token")
+			isLibraryStr, _ := cmd.Flags().GetString("library")
+			isLibraryStr = strings.ToLower(strings.TrimSpace(isLibraryStr))
+			isLibrary := false
+			if isLibraryStr == "y" || isLibraryStr == "yes" {
+				isLibrary = true
+			}
+			_, err = repository.GenerateGithub(nameOfApp, "", token, isLibrary)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+}
+
+func generateApplication() *cobra.Command {
+	return &cobra.Command{
+		Use:   "generate-project",
+		Short: "Generates the boilerplate for a given project type",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+			var cloneUrl string
+			ctx := context.Background()
+			nameOfApp, _ := cmd.Flags().GetString("name")
+			goVer, _ := cmd.Flags().GetString("go")
+			projectLocation, _ := cmd.Flags().GetString("project-location")
+			projType, _ := cmd.Flags().GetString("type")
+			createRepositoryInput, _ := cmd.Flags().GetString("create-repository")
+			createRepositoryInput = strings.ToLower(strings.TrimSpace(createRepositoryInput))
+			// Can't create repo unless project type has been provided in a flag, so prompt user for it
+			if createRepositoryInput == "y" || createRepositoryInput == "yes" {
+				err = projectgeneration.ValidateMandatoryArguments(nameOfApp, projType, projectLocation, goVer)
+				if err != nil {
+					log.Event(ctx, "Error generating a complete project", log.Error(err))
+					return err
+				}
+				isLibrary := false
+				cloneUrl, err = repository.GenerateGithub(nameOfApp, projectgeneration.ProjectType(projType), "", isLibrary)
+				isEmptyDir, err := projectgeneration.IsEmptyDir(projectLocation + "/" + nameOfApp)
+				if err != nil {
+					log.Event(ctx, "error unable to validate project location", log.Error(err))
+					return err
+				}
+				if isEmptyDir {
+					projectgeneration.OfferPurgeProjDestination(ctx, projectLocation, nameOfApp)
+				}
+				repository.CloneRepository(ctx, cloneUrl, projectLocation, nameOfApp)
+
+				err = projectgeneration.GenerateProject(nameOfApp, projType, projectLocation, goVer, true)
+				if err != nil {
+					return err
+				}
+
+				repository.PushToRepo(ctx, projectLocation, nameOfApp)
```

Comment:
consider adding a return after PushToRepo so the alternative path doesn't have to be nested in an else block